module/zfs/zpl_inode.c

```diff
@@ -353,6 +353,27 @@ zpl_getattr_impl(const struct path *path, struct kstat *stat, u32 request_mask,
 	 */
 
 	error = -zfs_getattr_fast(path->dentry->d_inode, stat);
+
+#ifdef STATX_BTIME
+	ZFS_TIME_DECODE(zp->z_crtime, btime);
+	stat->result_mask |= STATX_BTIME;
+#endif
+
+#ifdef STATX_ATTR_IMMUTABLE
```

Comment:
Similar to the `result_mask`, there is an `attributes_mask`. You need to always set `stat->attributes_mask | = STATX_ATTR_IMMUTABLE` to indicate that you are returning an answer for that flag.
```
#ifdef STATX_ATTR_IMMUTABLE
    stat->attributes_mask |= STATX_ATTR_IMMUTABLE;
	if (zfs_flags & ZFS_IMMUTABLE)
		stat->attributes |= STATX_ATTR_IMMUTABLE;
#endif
```

Likewise for the other flags. Alternatively, if you're willing to skip the idea of per-flag #ifdef, you could set the mask once, which is what the other filesystems do:
```
stat->attributes_mask |= (STATX_ATTR_APPEND |
                          STATX_ATTR_IMMUTABLE |
                          STATX_ATTR_NODUMP);
```

Reply:
That makes more sense than my cursory read of the interface. It will be fixed in the next push.